Sources/NIOKit/NIOKitTestCase.swift

```diff
@@ -0,0 +1,35 @@
+import NIO
+import XCTest
+
+protocol NIOKitTestCaseProt: class {
+    var group: EventLoopGroup! { get set }
+}
+
+extension NIOKitTestCaseProt {
+    internal var eventLoop: EventLoop {
+        return self.group.next()
+    }
+    
+    internal func setupGroup() {
+        self.group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+    }
+    
+    internal func teardownGroup() {
+        XCTAssertNoThrow(try self.group.syncShutdownGracefully())
+        self.group = nil
+    }
+}
+
+class NIOKitTestCase: XCTestCase, NIOKitTestCaseProt {
```

Comment:
Even if the protocols are internal, this class should be public. You might have to make the protocols public to make this work, but I'm not sure.

Reply:
Oh, yeah, totally forgot that. I'll publicise all the things :D